controller/src/main/java/org/jboss/as/controller/ServiceVerificationHelper.java

```diff
@@ -120,7 +122,13 @@ public synchronized void execute(final OperationContext context, final ModelNode
             reportImmediateDependants(problemList, failureDescription);
 
             if (context.isRollbackOnRuntimeFailure()) {
-                context.setRollbackOnly();
+                context.attach(DEFFERED_ROLLBACK_ATTACHMENT, Boolean.TRUE);
```

Comment:
Hi @spyrkob -- can you add a short comment above this line explaining what this is about? Thanks.

Other than that, LGTM.

Reply:
Thanks @bstansberry , I added the comment above.